components/badge/badge_eink_dev.c

```diff
@@ -190,42 +213,31 @@ badge_eink_dev_init(enum badge_eink_dev_t dev_type)
 	if (res != ESP_OK)
 		return res;
 
-	gpio_matrix_out(PIN_NUM_EPD_MOSI, VSPID_OUT_IDX, 0, 0);
-	gpio_matrix_out(PIN_NUM_EPD_CLK, VSPICLK_OUT_IDX, 0, 0);
-	gpio_matrix_out(PIN_NUM_EPD_CS, VSPICS0_OUT_IDX, 0, 0);
-	CLEAR_PERI_REG_MASK(SPI_SLAVE_REG(SPI_NUM), SPI_TRANS_DONE << 5);
-	SET_PERI_REG_MASK(SPI_USER_REG(SPI_NUM), SPI_CS_SETUP);
-	CLEAR_PERI_REG_MASK(SPI_PIN_REG(SPI_NUM), SPI_CK_IDLE_EDGE);
-	CLEAR_PERI_REG_MASK(SPI_USER_REG(SPI_NUM), SPI_CK_OUT_EDGE);
-	CLEAR_PERI_REG_MASK(SPI_CTRL_REG(SPI_NUM), SPI_WR_BIT_ORDER);
-	CLEAR_PERI_REG_MASK(SPI_CTRL_REG(SPI_NUM), SPI_RD_BIT_ORDER);
-	CLEAR_PERI_REG_MASK(SPI_USER_REG(SPI_NUM), SPI_DOUTDIN);
-	WRITE_PERI_REG(SPI_USER1_REG(SPI_NUM), 0);
-	SET_PERI_REG_BITS(SPI_CTRL2_REG(SPI_NUM), SPI_MISO_DELAY_MODE, 0,
-			SPI_MISO_DELAY_MODE_S);
-	CLEAR_PERI_REG_MASK(SPI_SLAVE_REG(SPI_NUM), SPI_SLAVE_MODE);
-
-	WRITE_PERI_REG(SPI_CLOCK_REG(SPI_NUM),
-			(1 << SPI_CLKCNT_N_S) | (1 << SPI_CLKCNT_L_S)); // 40 MHz
-//	WRITE_PERI_REG(SPI_CLOCK_REG(SPI_NUM), SPI_CLK_EQU_SYSCLK); // 80Mhz
-
-	SET_PERI_REG_MASK(SPI_USER_REG(SPI_NUM),
-			SPI_CS_SETUP | SPI_CS_HOLD | SPI_USR_MOSI);
-	SET_PERI_REG_MASK(SPI_CTRL2_REG(SPI_NUM),
-			((0x4 & SPI_MISO_DELAY_NUM) << SPI_MISO_DELAY_NUM_S));
-	CLEAR_PERI_REG_MASK(SPI_USER_REG(SPI_NUM), SPI_USR_COMMAND);
-	SET_PERI_REG_BITS(SPI_USER2_REG(SPI_NUM), SPI_USR_COMMAND_BITLEN, 0,
-			SPI_USR_COMMAND_BITLEN_S);
-	CLEAR_PERI_REG_MASK(SPI_USER_REG(SPI_NUM), SPI_USR_ADDR);
-	SET_PERI_REG_BITS(SPI_USER1_REG(SPI_NUM), SPI_USR_ADDR_BITLEN, 0,
-			SPI_USR_ADDR_BITLEN_S);
-	CLEAR_PERI_REG_MASK(SPI_USER_REG(SPI_NUM), SPI_USR_MISO);
-	SET_PERI_REG_MASK(SPI_USER_REG(SPI_NUM), SPI_USR_MOSI);
-
-	char i;
-	for (i = 0; i < 16; i++) {
-		WRITE_PERI_REG((SPI_W0_REG(SPI_NUM) + (i << 2)), 0);
-	}
+	spi_bus_config_t buscfg = {
+        .miso_io_num = -1,  // MISO not used, we are transferring to the slave only
+        .mosi_io_num = PIN_NUM_EPD_MOSI,
+        .sclk_io_num = PIN_NUM_EPD_CLK,
+        .quadwp_io_num = -1,
+        .quadhd_io_num = -1,
+        // The maximum size sent below covers the case
+        // when the whole frame buffer is transferred to the slave
+        .max_transfer_sz = DISP_SIZE_X_B * DISP_SIZE_Y,
```

Comment:
does this mean that the esp-idf code does an extra allocate of this amount of memory? If that's the case, then it might be better to use blocks of 256 bytes, or even less.

Reply:
Resolved in #214